### PR TITLE
Optionally read tweet id’s from the standard input during hydration.

### PR DESCRIPTION
**Pull request by dimazest**
This is a proof of concept implementation.

The good points:

* it can read archives
* it also reads from the std input.

The not so good points:

* the code is a bit ugly because I tried to eliminate an inline `open()`.

### CODE REVIEW

**Comment by edsu** [comment]
This is beautiful, thank you! The only problem I'm having is how do you get twarc.py to read the ids from stdin? 

```
cat ids.txt | twarc.py --hydrate > tweets.json
usage: twarc [-h] [-v] [--search SEARCH] [--max_id MAX_ID]
             [--since_id SINCE_ID] [--result_type {mixed,recent,popular}]
             [--lang LANG] [--track TRACK] [--follow FOLLOW]
             [--locations LOCATIONS] [--sample] [--timeline TIMELINE]
             [--timeline_user_id TIMELINE_USER_ID]
             [--lookup_screen_names LOOKUP_SCREEN_NAMES [LOOKUP_SCREEN_NAMES ...]]
             [--lookup_user_ids LOOKUP_USER_IDS [LOOKUP_USER_IDS ...]]
             [--hydrate HYDRATE] [--log LOG] [--consumer_key CONSUMER_KEY]
             [--consumer_secret CONSUMER_SECRET] [--access_token ACCESS_TOKEN]
             [--access_token_secret ACCESS_TOKEN_SECRET] [-c CONFIG]
             [-p PROFILE] [-w]
twarc: error: argument --hydrate: expected one argument
```

**Comment by edsu** [comment]
Oh I see now: 

    cat ids.txt | twarc.py --hydrate - > tweets.json

File: twarc.py [hunk 1]
```diff
@@ -140,7 +141,16 @@ def main():
         tweets = t.filter(track=args.track, follow=args.follow,
                           locations=args.locations)
     elif args.hydrate:
-        tweets = t.hydrate(open(args.hydrate, 'rU'))
+        f_names = [args.hydrate] if args.hydrate != '--hydrate' else None
```
Comment (edsu) [inline twarc.py]: @dimazest can you tell me what this line is doing?

**Comment by edsu** [comment]
i made a few changes and pulled them into master. Thanks for pushing this forward!

File: twarc.py (hunk 1 above)
Comment (dimazest) [inline twarc.py]: The idea was to read from stdin if `-` is passed to pipe ids into twarc:

```
zcat ids.csv.gz | twarc.py --hydrate -
```

but in this case `args.hydrate` will somehow be equal to `--hydrate`.

Comment (edsu) [inline twarc.py]: I wee. Well I've tested the use of '--hydrate -' with what is now on master and it works fine. Thanks very much for your contribution @dimazest

**Pull request merged**